docs/route-tasks.md

```diff
@@ -0,0 +1,138 @@
+- A Worker runs on a speicific host. This means that Activity Tasks that depend on resources specific to a particular host must be executed by a Worker on that host.
+
+So, for any given Temporal Workflow, how do we make sure a specific Worker is executing a specific Task? 
+
+## Use Task Queues
+
+"Task Routing" is the development technique that makes use of Task Queues to orchestrate the execution of Tasks by specific Workers.
+
+Let's look at the ["Hello World" sample in Go](https://github.com/temporalio/temporal-go-samples/tree/master/helloworld) to see how this is done.
+
+Here, we are creating a new Worker that is subscribed to the "hello-world" Task Queue, and is registered to handle the `helloworld.Workflow` and `helloworld.Activity`:
+
+```go
+// The Worker is created and subscribed to the "hello-world" Task Queue
+w := worker.New(c, "hello-world", worker.Options{})
+defer w.Stop()
+// The Worker is registered to handle the helloworld Workflow
+w.RegisterWorkflow(helloworld.Workflow)
+// The Worker is registered to handle the helloworld Activity
+w.RegisterActivity(helloworld.Activity)
+```
+
+And here, we are starting the `helloworld.Workflow`:
+
+```go
+// The Workflow Options specify the "hello-world" Task Queue
+// This is the Task Queue that the Workflow will add its Tasks to
+workflowOptions := client.StartWorkflowOptions{
+  ID:        "hello_world_workflowID",
+  TaskQueue: "hello-world",
+}
+// The Workflow Options are passed to the execution of the Workflow
+we, err := c.ExecuteWorkflow(
+  context.Background(), 
+  workflowOptions, 
+  helloworld.Workflow, 
+  "Temporal",
+)
+```
+
+Task routing can be as simple as making sure a Worker is subscribed to a Task Queue and Tasks are being added to it.
+
+:::caution
+
+If a Worker pulls a Task that it has not been registered to run, an "unknown workflow type" error will occur.
+
+:::
+
+## Task routing for Activity dependencies
+
+The Temporal service will persist Activity results as long as the results can be serialized and stored in a Cassandra or MySQL database (which happens to be true for most Activity results). However, there are some scenarios where an Activity execution will do something that results in data being stored on the host where the Worker is running. It is also possible that the next Activity in the Workflow has a dependency on that data.
+
+Let's look at a file processing Workflow, as an example,  where files are downloaded, processed in some way, and then uploaded to some other location. If a large video file is being processed only the name of the file can persist normally. The file itself must persist locally on the Worker host. This means that any future Activities that act on the video file must also take place on the host where the Worker downloaded it.
+
+So, how do we ensure that a Worker on the correct host will execute the next set of Activities that depends on the video file?
+
+Each SDK offers a slightly different way to handle this.
```

Comment:
"Each SDK" is sort of a negative reflection on our product. I would say, "This is accomplished in different ways depending on the SDK".